### DISCUSSION TITLE
Enter own API key in settings

### DISCUSSION THREAD

**Discussion by michael9dk** [Docile-Alligator/Infinity-For-Reddit#1511] [Ideas]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/categories/ideas) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/categories/completed-ideas) ideas to see if someone else has already submitted the same request.
- [X] I will describe the feature with as much detail as possible.

### Description

If Infinity had an option to enter API key (set redirect-URL, and User-agent generated from random words) while logging in the first time, it would make it easier for users to use their own API key. 

Benefits:
- You can publish a key-less build in PlayStore and Fdroid.
- Users wont need to build.
- Ensure Infinity's redirect url/user agent/name isn't used in personal builds.
- Users can select to use their own key or subscribe.
- Infinity's private API key can be auto-loaded, when users have a valid subscription.
- One code base to maintain, with proper use of Play and key store.

### Additional context

Feature request is inspired by this sub.

https://www.reddit.com/r/Infinity_For_Reddit/comments/14c7v84/if_you_want_to_use_your_own_api_key/

### How important is this feature?

Important

**Comment by demirdegerli** [comment]
This is not possible legally.
![IMG_20230627_134923](https://github.com/Docile-Alligator/Infinity-For-Reddit/assets/67432156/6d076e64-c296-4862-b187-04a489be03c6)

**Reply by Magneticdud** [reply]
That only applies to something published on the play store.

On something open source that you need to compile by yourself (or having fdroid compile for you), why not?

**Reply by eiqnepm** [reply]
Because it would go against the terms of Reddit and they'd most likely resort to banning accounts that used the app.

**Reply by Magneticdud** [reply]
Why? Using a personal developer key for a new app still need to pay the insane fees to them

(Unless you mean to insert the official Reddit app API key)

**Reply by rpdelaney** [reply]
As a user, it should be my choice to accept the risk that I could get banned or not. For my part, if they ban me, fine. I'd much prefer not using reddit to using the official app.

**Reply by rpdelaney** [reply]
By the way, don't take legal advice from your adversary.

**Comment by demirdegerli** [comment]
@KhoalaS made a fork 🎉
https://github.com/KhoalaS/Infinity-For-Reddit

**Reply by 1RandomDev** [reply]
That app looks really promising. And it supports all things that Infinity didn't want to add like messaging and using unofficial API endpoints.

**Reply by KaKi87** [reply]
Unfortunately it's dead

**Comment by ghost** [comment]
Why shouldn't we be able to enter the key ourselves? What's the difference between compiling a separate app with our personal key and entering that same key in a complete app that supports the feature? Functionally nothing. It achieves the same thing but the latter has has benefits to the users and the devs:

- Users won't have to waste time trying to compile the custom .apk with bs Google cloud scripting.
- They won't have to trust the maker of the script or Google to not backdoor the .apk.
- Users wouldn't have to have a Google account in order to use Google's compiler.
- Users who would want to use both the official and compiled app on the same device (one for anonymous usage, the other for the account) cannot do this currently, because of the same app id. This applies to work- and secondary user profiles.
- Instead of making forks with separate app id's and finctionalities supporting this feature natively would prevent breaking up the development into smaller projects.

**Reply by ghost** [reply]
Also I think it's worth reevaluating how much do we want to trust/respect Reddit's say in this in terms of their Policies. Because as far as I can say they don't give a f neither about us nor the devs.
